### PR TITLE
improvement: add request id to logs

### PR DESCRIPTION
**Pull request by hails**
## Description
This PR adds `req.id` to every log to make tracking easier.
Also, makes use of `X-Request-ID` header as default log id.

## Your checklist for this pull request

:rotating_light: Please review this items for a good pull request. :four_leaf_clover:

1. I've read the project's [Contributing Guidelines](../CONTRIBUTING.md)
1. My commits are well written and follow [pagarme/git-style-guide](https://github.com/pagarme/git-style-guide)
1. My changes are well covered by **tests** and **logs**
1. I've updated the project docs (if needed)
1. I fell safe about this implementation
1. I feel comfortable with the code I wrote, and I'm not ashamed to show it to my friends

In a good pull request, everything above is true :relaxed:

### CODE REVIEW

File: src/controllers/receipt.js [hunk 1]
```diff
@@ -26,11 +27,12 @@ const getLastReceipt = (receiptId) => {
 const show = (req, res) => {
   const receiptId = req.params.receipt_id
 
-  return getLastReceipt(receiptId)
+  return getLastReceipt(receiptId, req.id)
```
Comment (wilkmaia) [inline src/controllers/receipt.js]: `loggerId = req.id`

File: src/controllers/receipt.js [hunk 2]
```diff
@@ -25,12 +26,14 @@ const getLastReceipt = (receiptId) => {
 
 const show = (req, res) => {
   const receiptId = req.params.receipt_id
+  const loggerId = req.get('X-Request-ID') || req.id
```
Comment (wilkmaia) [inline src/controllers/receipt.js]: I believe this should be `x-request-id`. We had a similar issue with this on another service. This should be tested before deploying.

File: src/controllers/receipt.js [hunk 3]
```diff
@@ -52,18 +56,21 @@ const show = (req, res) => {
...
         receiptId,
         err,
+        id: loggerId,
       })
     })
 }
 
 const render = (req, res) => {
   const receiptId = req.params.receipt_id
+  const loggerId = req.get('X-Request-ID') || req.id
```
Comment (wilkmaia) [inline src/controllers/receipt.js]: I believe this should be `x-request-id` (small caps). We had a similar issue with this on another service. This should be tested before deploying.

File: src/controllers/receipt.js (hunk 2 above)
Comment (alalbux) [reply src/controllers/receipt.js]: @wilkmaia It's right, it's `X-Request-ID`.

File: src/controllers/receipt.js (hunk 3 above)
Comment (alalbux) [reply src/controllers/receipt.js]: it's `X-Request-ID`

**Pull request merged**